docs/src/app/(private)/experiments/custom-components.tsx

```diff
@@ -0,0 +1,78 @@
+'use client';
+import * as React from 'react';
+import { useRenderer } from '@base-ui-components/react/use-renderer';
+import styles from './custom-components.module.css';
+
+type TextProps = {
+  weight?: 'light' | 'regular' | 'bold';
+  className: any;
+  render?: any;
```

Comment:
How would users type this properly? Like this?

```suggestion
  render?: useRenderer.Settings<any, Element>['render'];
```

Reply:
We'd export a specific type for render as Michal commented: https://github.com/mui/base-ui/pull/1418#discussion_r1944835602

Reply:
I've exported a RenderProp<State> type, and I've updated the example to use all the right types.